hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,317 @@
+---
```

Comment:
sorry for being a bit late Martin.
I read through all of this again and it does make sense to me.  Although I have to admit the way the duty cycle resolution is used was a bit confusing to me at first.  But maybe this description is more friendly to software? Because otherwise software would need to do all the work to convert between time / count anyways. 

thanks!